src/main/java/io/stargate/sgv2/jsonapi/service/embedding/configuration/PropertyBasedEmbeddingServiceConfig.java

```diff
@@ -1,40 +1,94 @@
 package io.stargate.sgv2.jsonapi.service.embedding.configuration;
 
+import com.fasterxml.jackson.annotation.JsonProperty;
 import io.smallrye.config.ConfigMapping;
 import io.smallrye.config.WithConverter;
 import io.smallrye.config.WithDefault;
 import jakarta.annotation.Nullable;
-import java.net.URL;
+import java.util.List;
+import java.util.Map;
 import java.util.Optional;
 import org.eclipse.microprofile.config.spi.Converter;
 
-@ConfigMapping(prefix = "stargate.jsonapi.embedding.service")
+@ConfigMapping(prefix = "jsonapi.vector.provider")
 public interface PropertyBasedEmbeddingServiceConfig {
 
   @Nullable
-  ServiceConfig openai();
```

Comment:
It may be late to ask this, but why wouldn't it just be something like:

    Map<String, ServiceConfig> providers

instead of hard-coding provider names as fields?

Reply:
I am not sure I understand the question correctly. I think that's the way for mapping configuration to objects. In the configuration yaml, I have something like:
```
jsonapi:
  vector:
    provider:
       openai:
       huggingface:
```
In the interface, I use `@ConfigMapping(prefix = "jsonapi.vector.provider")` to locate the configuration, and the field name `VectorProviderConfig openai()` will be mapped to `openai` in the configuration file.